internal/rpc/models.go

```diff
@@ -34,27 +35,132 @@ var reactFormRPC = db.Code{
+    return {"schema" : schema, "uiSchema" : uiSchema}
 
-def makeResponse(input):
+def regular(fieldName, storedAs, datatype):
+    type = FindOne("enumValue", Eq("id",  storedAs)).GetString("name")
+    if   type == "bool":
+          type = "boolean"
+    elif type == "int":
+         type = "integer"
+    elif type == "float":
+         type = "number"
+    else:
+        type  = "string"
     return {
-       "title"       : input,
-       "description" : "Use form to add new model.",
-       "type"        : "object",
-       "properties"  : getDatatypes(input)
+        "type" : type, 
+        "title" : fieldName, 
+        "datatype" : datatype
     }
-    
-input = args["model"]
-result(makeResponse(input))`,
+
+def enum(fieldName, id, datatype):
+    evs = FindMany("enumValue", EqFK("datatype", id))
+    evn = []
+    evi = []
+    for ev in evs:
+        evn.append(ev.Get("name"))
+        evi.append(ev.Get("id"))
+    return {
+        "type" : "string", 
+        "title" : fieldName, 
+        "enum": evi, 
+        "enumNames": evn, 
+        "datatype" : datatype
+    }
+
+def ui(type):
+    if type == "emailAddress":
+        return {
+            "ui:options": { "inputType": "email"}
+        }
+    elif type == "url":
+        return {"ui:widget" : "uri", "ui:placeholder": "http://"}
+    elif type == "bool":
+        return {"ui:widget" : "select"}
+    elif type == "password":
+        return {"ui:widget" : "password"}
+    elif type == "longText":
+        return {"ui:widget" : "textarea"}
+    elif type == "phone" :
+        return {
+            "ui:options": { "inputType": "tel"}
+        }
+    return None
+
+out = process(args["model"])
+result({"schema" : {
+       "type"        : "object",
+       "properties"  : out["schema"]
+    },
+       "uiSchema" : out["uiSchema"]})`,
+}
+
+var validateRPC = db.Code{
+	ID:                db.MakeID("d7633de5-9fa2-4409-a1b2-db96a59be52b"),
+	Name:              "validate",
+	Runtime:           db.Starlark,
+	FunctionSignature: db.RPC,
+	Code: `def validate(args):
+    properties = args["schema"]["properties"]
+    data = args["data"]
+    errors = {}
+    for name in properties:
+        x = FindOne("datatype", Eq("name", properties[name]["datatype"]))
+        if x.GetBool("enum") == False:
+            y = FindOne("code", EqID(x.GetFK("validator")))
+        else:
+            y = FindOne("code", Eq("name", "uuid"))
+        inp = ""
+        if name in data:
+            inp = str(data[name])
+        out, ran = Exec(y, inp)
+        #If there is an error from a validator
+        if ran == False:
+            errors[name] = {"__errors" : [out]}
+    return errors
+        
+result(validate(args))`,
 }
+
+var replRPC = db.Code{
+	ID:                db.MakeID("591bc8f7-543b-4fa9-bdf7-8948c79cdd26"),
+	Name:              "repl",
+	Runtime:           db.Starlark,
+	FunctionSignature: db.RPC,
+	Code: `def repl(args):
+    out, ran = Exec(args["data"], "")
```

Comment:
I realize we're early here, but I think we should really consider the security implications (alarm bells!!) of having something like this in the code.

Reply:
I realize the REPL is basically just "SQL Injection as a Feature", so what I have in mind here is at least some kind of authentication -- which I realize you're blocked on me for :p 

so maybe just add some big scary comments around this for now!